src/modal-template.ts

```diff
@@ -0,0 +1,244 @@
+
+import { ModalConfig } from './modal-config';
+
+@customElement('modal-template')
+export class ModalTemplate extends LitElement {
+  /**
+   * The ModalConfig that displayed the template
+   *
+   * @type {ModalConfig}
+   * @memberof ModalTemplate
+   */
+  @property({ type: Object }) config: ModalConfig = new ModalConfig();
+
+  /** @inheritdoc */
+  render(): TemplateResult {
+    return html`
+      <div class="modal-wrapper">
+        <div class="modal-container">
+          <header style="background-color: ${this.config.headerColor}">
+            ${this.config.showCloseButton ? this.closeButtonTemplate : ''}
+            <div class="logo-icon">
+              <img
+                src="https://archive.org/images/logo_arches.png"
+                alt="Internet Archive Logo"
+              />
+            </div>
+            ${this.config.title
+              ? html`<h1 class="title">${this.config.title}</h1>`
+              : ''}
+            ${this.config.subtitle
+              ? html`<h2 class="subtitle">${this.config.subtitle}</h2>`
+              : ''}
+          </header>
+          <section class="modal-body">
+            <div class="content">
+              <div
+                class="processing-logo ${this.config.showProcessingIndicator
+                  ? ''
+                  : 'hidden'}"
+              >
+                <ia-activity-indicator
+                  .mode=${this.config.processingImageMode}
+                ></ia-activity-indicator>
+              </div>
+
+              ${this.config.headline
+                ? html` <h1 class="headline">${this.config.headline}</h1> `
+                : ''}
+              ${this.config.message
+                ? html` <p class="message">${this.config.message}</p> `
+                : ''}
+
+              <div class="slot-container">
+                <slot> </slot>
+              </div>
+            </div>
+          </section>
+        </div>
+      </div>
+    `;
+  }
+
+  /**
+   * Dispatch the `closeButtonPressed` event to the consumer
+   *
+   * @private
+   * @memberof ModalTemplate
+   */
+  private handleCloseButton(): void {
+    const event = new Event('closeButtonPressed');
+    this.dispatchEvent(event);
+  }
+
+  /**
+   * The close button template
+   *
+   * @readonly
+   * @private
+   * @type {TemplateResult}
+   * @memberof ModalTemplate
+   */
+  private get closeButtonTemplate(): TemplateResult {
+    return html`
+      <button
+        type="button"
+        class="close-button"
+        @click=${this.handleCloseButton}
+      >
+        ${closeIcon}
+      </button>
+    `;
+  }
+
+  /** @inheritdoc */
+  static get styles(): CSSResult {
+    const processingImageSize = css`var(--processingImageSize, 75px)`;
+
+    const modalCornerRadius = css`var(--modalCornerRadius, 10px)`;
+    const modalBorder = css`var(--modalBorder, 2px solid black)`;
+    // if the content of the modal is too big to fit on screen, this sets the bottom margin
```

Comment:
You could alternately set a `max-height` of something like `calc(100vh - 75px)` and set `overflow: auto` to allow the contents to scroll.

Reply:
So this is interesting.. the `calc(100vh - 75px)` calculates the initial size correctly, but then doesn't change after that so if you resize your window, the height doesn't change.

For some reason that `window.onresize` event listener that is currently setting the `--containerHeight` causes the `calc()` to recalculate, even if I set any random CSS variable. If I remove that `onresize` the `calc()` stops working.

Reply:
That's...odd. I thought any value with a relative measurement would recalculate with a repaint pass, like when the viewport resizes. Maybe it's calculating a pixel value because of the mixed units?

Reply:
I'm not sure if this would have any effect, but since this is happening inside the shadow root, do you know if that affects these calculations? That was actually the original reason I put in that `window.onresize` ... I couldn't figure out a way to properly calculate the content's frame since it didn't have access to anything above its own DOM.

Reply:
`window` is the same in shadow and light DOM, so I would imagine the same issue should present itself in a plain static page.

Reply:
https://output.jsbin.com/mekezup behaves as expected, so maybe it is a shadow DOM issue.

Reply:
https://output.jsbin.com/wenoquh/1 .. I took the same code you used, put it inside the shadowDOM and I'm seeing similar behavior where it renders properly at first, but then doesn't resize.

Reply:
Sooo this looks like it's only an issue in Safari... I tested that in Firefox and Chrome and both were resizing properly

Reply:
I changed it to use the `calc(100vh .. )` calculation and that works nicely. In Firefox and Chrome, it updates as you resize. In Safari, there's a 100ms throttler to trigger the re-render.